lib/uri_parser.rb

```diff
@@ -20,7 +20,11 @@ def query_params
       {}.tap do |hash|
         k_v_pairs.each do |kv|
           key, value = kv.split('=')
-          hash[CGI.unescape(key)] = value && CGI.unescape(value)
+          key = CGI.unescape(key) if !key.nil?
+          v = CGI.unescape(value) if !value.nil?
+          if !key.nil? || !value.nil?
```

Comment:
(nit)
Consider using [Hash#compact!](https://www.geeksforgeeks.org/ruby-hash-compact-function/) to remove nils instead of doing it yourself

Reply:
```
hash[CGI.unescape(key)] = value && CGI.unescape(value) if key
```
should work too (if key is nil, so is value)

Reply:
Looks like `compact!` was added in ruby 2.4 and this uses ruby 2.3 :/

Reply:
[Hash#reject!](https://ruby-doc.org/core-2.3.0/Hash.html#method-i-reject-21) might work,then?

Reply:
Looks like `Hash.reject!` works